### PR TITLE
feat(entities): add entities-vaults package

### PR DESCRIPTION
**Pull request by Leopoldthecoder**
# Summary

Adding shared vault components.

## PR Checklist

* [ ] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [ ] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

File: packages/entities/entities-vaults/src/components/VaultConfigCard.cy.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import { konnectCardConfig, vault } from '../../fixtures/mockData'
+import VaultConfigCard from './VaultConfigCard.vue'
+import { EntityBaseConfigCard } from '@kong-ui-public/entities-shared'
+
+describe('<UpstreamsConfigCard/>', () => {
```
Comment (raoxiaoyan) [inline packages/entities/entities-vaults/src/components/VaultConfigCard.cy.ts:5]: `UpstreamsConfigCard` maybe  is `VaultConfigCard`

**Review by raoxiaoyan — APPROVED** [review]
LGTM

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -54,7 +51,12 @@ Cypress.Commands.add('mount', (component, options = {}) => {
        * (This is why this is disabled for now)
        *
        */
-      // app.use(options.router)
+      if (options.router) {
+        app.use(options.router)
+      } else {
+        options.router = defaultRouter
+        options.global.components.RouterLink = RouterLink
+      }
```
Comment (adamdehaven) [inline cypress/support/commands.ts:59]: question: just verifying this was copied from the private repo?

File: package.json [hunk 3]
```diff
@@ -72,6 +73,7 @@
     "typedoc": "^0.24.8",
     "typedoc-plugin-markdown": "^3.15.4",
     "typescript": "~5.1.6",
+    "uuid": "^9.0.0",
```
[the remark this answers is not in the document]
Comment (Leopoldthecoder) [reply package.json:76]: This is also from the private repo. My guess is other entity packages are also using it in their tests.

File: cypress/support/commands.ts (hunk 2 above)
Comment (Leopoldthecoder) [reply cypress/support/commands.ts:59]: Kind of, but in the private repo it is
```ts
if (options.router) {
  app.use(options.router)
} else {
  app.use(defaultRouter) // different here
  options.global.components.RouterLink = RouterLink
}
```

If I use `app.use(defaultRouter)` here, [this test](https://github.com/Kong/public-ui-components/blob/3ab908c4b5250e67dcbe36a834d394b60ba3be6e/packages/portal/spec-renderer/src/components/SpecRenderer.cy.ts#L85) will fail. Considering it was `options.router = defaultRouter` [in the first place](https://github.com/Kong/public-ui-components/blob/3ab908c4b5250e67dcbe36a834d394b60ba3be6e/cypress/support/commands.ts#L38), I kept it as is.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (adamdehaven) [reply package.json:76]: Ok just making sure it's only for tests, etc. Sounds good

**Review by adamdehaven — APPROVED** [review]
Code LGTM, did not functionally test

**Comment by Leopoldthecoder** [comment]
Functionally verified [here](https://github.com/Kong/kong-admin/pull/2834) (thank you @sumimakito) with all tests passing. Merging this now🚀.

**Comment by sumimakito** [comment]
Also manually tested with Kong Manager and LGTM 🚀

**Pull request merged**